strapi/strapi
# Make users-permissions auth strategy use the content API permissions engine

### What does it do?

- **Authentication:** users-permissions auth strategy now generate an ability using the content API permission engine instance and pass it down to the auth context for every authenticated request
- **Authorization:** users-permissions auth strategy now use the previously generated ability to check if the request can be allowed or not 

### Why is it needed?

It's the first step in harmonizing every content API auth layer by using the shared content API permission engine to make the checks. 

### How to test it?

Try to make authenticated (and non-authenticated) request to the content API as usual; nothing should've changed. This PR should only change the implementation, not the behavior.
